src/services/share-buttons.service.ts

```diff
@@ -1,9 +1,7 @@
 import { Injectable, EventEmitter } from '@angular/core';
 import { Http, Jsonp, Headers, RequestOptions } from '@angular/http';
 
-import { Observable } from 'rxjs/Observable';
-import 'rxjs/add/operator/catch';
-import 'rxjs/add/Observable/empty';
+import { Observable } from 'rxjs';
```

Comment:
I believe [tree shaking](https://angular.io/docs/ts/latest/cookbook/aot-compiler.html#!#tree-shaking) prevents angular from importing the whole library. It'll just import what is used. 

Correct me if I'm wrong please